### PR TITLE
fix to eq functions for naive models per #1812

### LINKED ISSUE #1812: Bug in tune model for snaive and grand_means in time series
You may have to override `eq_function` attr to make sure that it doesn't clash with the regular naive forecaster

_Originally posted by @Yard1 in https://github.com/pycaret/pycaret/pull/1806#discussion_r744179846_

In tune models, since the base class is the same for naive, snaive and grand_means, when we tune snaive and grand_means, it uses the grid from naive instead of using their own respective grids.

### PR DESCRIPTION
**Pull request by ngupta23**
Closes #1812

#### Describe the changes you've made

Added eq_function to model containers for `naive`, `snaive` and `grand_means` model.

## Type of change

Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit tests run locally

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/containers/models/time_series.py [hunk 1]
```diff
@@ -376,31 +410,23 @@ def __init__(self, globals_dict: dict) -> None:
...
-            tune_grid = {
-                "strategy": ["last", "mean", "drift"],
-                "sp": [1],
-                # Removing fh for now since it can be less than sp which causes an error
-                # Will need to add checks for it later if we want to incorporate it
-                "window_length": [None],  # , len(fh)]
-            }
+        tune_grid = {
+            "strategy": ["last", "mean"],
+            "sp": [self.sp, 2 * self.sp],
```
Comment (Yard1) [inline pycaret/containers/models/time_series.py:425]: Is it possible for sp to be 1 here?

Comment (ngupta23) [reply pycaret/containers/models/time_series.py:425]: Do I need to disable this model container when seasonality is 1?

Comment (Yard1) [reply pycaret/containers/models/time_series.py:425]: I think that would be best

**Review by Yard1 — APPROVED** [review]
Awesome!

**Pull request merged**